lib/axiom/core/event.js

```diff
@@ -45,6 +45,74 @@ export {AxiomEvent};
 export default AxiomEvent;
 
 /**
+ * The returned meta-event will fire as soon as (with opt_fireOnce) or whenever
+ * (with opt_fireOnce off) one of the sub-events fire.
+ *
+ * @param {!Array<!AxiomEvent>} events
+ * @param {boolean=} opt_fireOnce
+ * @return {!AxiomEvent}
+ */
+AxiomEvent.any = function(events, opt_fireOnce) {
+  var fireOnce = opt_fireOnce || false;
+  var event = new AxiomEvent();
+  
+  var listener = function(var_args) {
+    event.fire.apply(event, arguments);
+  };
+
+  for (var i = 0; i < events.length; ++i) {
+    if (fireOnce) {
+      events[i].listenOnce(listener);
+    } else {
+      events[i].addListener(listener);
+    }
+  }
+  
+  return event;
+};
+
+/**
+ * The returned meta-event will fire as soon as all of the sub-events have fired
+ * at least once. After that, the process either stops (with opt_fireOnce on)
+ * or resets (with opt_fireOnce on).
+ * 
+ * @param {!Array<!AxiomEvent>} events
+ * @param {boolean=} opt_fireOnce
+ * @return {!AxiomEvent}
+ */
+AxiomEvent.all = function(events, opt_fireOnce) {
+  var fireOnce = opt_fireOnce || false;
+  var event = new AxiomEvent();
+  var firedEvents = [];
+  var listeners = [];
+  
+  var listener = function(eventIdx, var_args) {
+    firedEvents[eventIdx] = true;
+    if (firedEvents.every(function(elt) { return elt === true; })) {
+      var args = Array.prototype.slice.call(arguments, 1);
+      event.fire.apply(event, args);
+      if (!fireOnce) {
+        for (var i = 0; i < events.length; ++i) {
```

Comment:
The stress here is on `events` being the source of truth: `fireEvents` is initialized from scratch, in the same way it's initially done in the setup code further below. I've just made it more explicit by  setting `firedEvents` to `[]` before the loop.